Uds 1793: Fix onclick in header navtree

davidornelas11 opened the pull request:
### Description
fixed navtree with onclick not getting called

<!-- Testing Steps -->
1. Add these props to a story `
navTree: [
  {
    text: "Home",
    selected: false,
    onClick: () => {
      console.log("Home clicked");
    }
  },
  {
    text: "Test 1",
    selected: false,
    onClick: () => {
      console.log("Test 1 clicked");
    }
  },
  {
    text: "Fun",
    selected: false,
    onClick: () => {
      console.log("Test 2 clicked");
    }
  }
],
  title: "Ira A. Fulton Schools of Engineering",
  parentOrg:
    "School of Computing, Informatics, and Decisions Systems Engineering",
  parentOrgUrl: "https://engineering.asu.edu",
  loggedIn: false,`
2. Check to make sure the oncall is called and that it is keyboard navigable

### Links

- [JIRA ticket](https://asudev.jira.com/browse/UDS-1793?atlOrigin=eyJpIjoiMDVkZDRkYjg0YzNhNDc3MmEyZWY3NzExODM1ZDUzMDIiLCJwIjoiaiJ9)

### FOR APPROVERS

- [Percy build approval](https://percy.io/5eae92d9/-all-UDS-packages)

davidornelas11 committed:
fix(component-header): fixed navtree with onclick not getting called

davidornelas11 committed:
test(component-header): add test to check for onclick call

If a navtree link does not have an href, check to make sure onclick is called

mlsamuelson reviewed (commented):
I tested with sublinks and they're not clickable by keyboard or mouse. The `tabindex="0"` isn't applying to the sublinks also, so they aren't focusable.

Per @scott-williams-az , it appears sublinks never supported onClick behaviors.

code I used for `navTree` to illustrate this:
```
const myNavTree = [
  {
    text: "Home",
    selected: false,
    onClick: () => {
      console.log("Home clicked");
    },
  },
  {
    text: "Test 1",
    selected: false,
    onClick: () => {
      console.log("Test 1 clicked");
    },
  },
  {
    text: "Fun",
    selected: false,
    onClick: () => {
      console.log("Test 2 clicked");
    },
  },
  {
    text: "Link option 1",
    href: "#",
    items: [
      [
        {
          text: "Sublink 1",
          selected: false,
          onClick: () => {
            console.log("Sublink 1 clicked");
          },
        },
        {
          href: "https://www.asu.edu/",
          text: "Sublink 2",
        },
      ],
    ],
  },
];
```

davidornelas11 committed:
fix(component-header): added onClick event to dropdown sublinks

davidornelas11 commented:
> I tested with sublinks and they're not clickable by keyboard or mouse. The `tabindex="0"` isn't applying to the sublinks also, so they aren't focusable.
> 
> Per @scott-williams-az , it appears sublinks never supported onClick behaviors.
> 
> code I used for `navTree` to illustrate this:
> 
> ```
> const myNavTree = [
>   {
>     text: "Home",
>     selected: false,
>     onClick: () => {
>       console.log("Home clicked");
>     },
>   },
>   {
>     text: "Test 1",
>     selected: false,
>     onClick: () => {
>       console.log("Test 1 clicked");
>     },
>   },
>   {
>     text: "Fun",
>     selected: false,
>     onClick: () => {
>       console.log("Test 2 clicked");
>     },
>   },
>   {
>     text: "Link option 1",
>     href: "#",
>     items: [
>       [
>         {
>           text: "Sublink 1",
>           selected: false,
>           onClick: () => {
>             console.log("Sublink 1 clicked");
>           },
>         },
>         {
>           href: "https://www.asu.edu/",
>           text: "Sublink 2",
>         },
>       ],
>     ],
>   },
> ];
> ```

I cleaned up the dropdown component a bit and added the onClick event

davidornelas11 committed:
test(component-header): add test for sublink  dropdown

mlsamuelson commented:
![Screenshot 2024-08-14 at 8 35 27 AM](https://github.com/user-attachments/assets/fb2fc428-f70f-46e2-918e-88c53bc0027b)

mlsamuelson commented:
Circular dependency: `DropdownItem/index.js` imports from `NavItem/index.js` and that in turn imports from `DropdownItem/index.js`. 

The DropdownItem component file is just pulling in the LINK_DEFAULT_PROPS, and we're getting other eslint errors about the props, so perhaps the DropdownItem component file just needs to do its own thing for the link props?

davidornelas11 committed:
style(component-header): fix linting errors

davidornelas11 committed:
fix(component-header): fixed prop that was not passed in properly

mlsamuelson reviewed (approved):
Issues resolved! Thanks, @davidornelas11
